File: frontend/src/types/stats.ts
export type StatsPeriod = 'weekly' | 'monthly';

export interface PeriodData {
  label: string;
  completed_count: number;
  total_count: number;
  completion_rate: number;
}

export interface CompletionStatsResponse {
  period: StatsPeriod;
  data: PeriodData[];
}

File: backend/schemas/stats.py
from typing import Literal

from pydantic import BaseModel


class PeriodData(BaseModel):
    label: str
    completed_count: int
    total_count: int
    completion_rate: float


class CompletionStatsResponse(BaseModel):
    period: Literal["weekly", "monthly"]
    data: list[PeriodData]


class CategoryStatData(BaseModel):
    category_id: str
    category_name: str
    total_count: int
    completed_count: int
    completion_rate: float


class CategoryStatsResponse(BaseModel):
    data: list[CategoryStatData]
